iphone/Maps/Core/iCloud/iCloudDocumentsDirectoryMonitor.swift

```diff
@@ -141,65 +141,34 @@ class iCloudDocumentsDirectoryMonitor: NSObject, CloudDirectoryMonitor {
     return metadataQuery
   }
 
-  class func getContentsFromNotification(_ notification: Notification, _ onError: (Error) -> Void) -> CloudContents {
+  class func getContentsFromNotification(_ notification: Notification) throws -> CloudContents {
     guard let metadataQuery = notification.object as? NSMetadataQuery,
           let metadataItems = metadataQuery.results as? [NSMetadataItem] else {
-      return []
+      throw SynchronizationError.failedToRetrieveMetadataQueryContent
     }
-
-    let cloudMetadataItems = CloudContents(metadataItems.compactMap { item in
-      do {
-        return try CloudMetadataItem(metadataItem: item)
-      } catch {
-        onError(error)
-        return nil
-      }
-    })
-    return cloudMetadataItems
+    return try metadataItems.map { try CloudMetadataItem(metadataItem: $0) }
   }
 
   // There are no ways to retrieve the content of iCloud's .Trash directory on the macOS because it uses different file system and place trashed content in the /Users/<user_name>/.Trash which cannot be observed without access.
   // When we get a new notification and retrieve the metadata from the object the actual list of items in iOS contains both current and deleted files (which is in .Trash/ directory now) but on macOS we only have absence of the file. So there are no way to get list of deleted items on macOS on didFinishGathering state.
   // Due to didUpdate state we can get the list of deleted items on macOS from the userInfo property but cannot get their new url.
-  class func getTrashContentsFromNotification(_ notification: Notification, _ onError: (Error) -> Void) -> CloudContents {
-    guard let removedItems = notification.userInfo?[NSMetadataQueryUpdateRemovedItemsKey] as? [NSMetadataItem] else { return [] }
-    return CloudContents(removedItems.compactMap { metadataItem in
-      do {
-        var item = try CloudMetadataItem(metadataItem: metadataItem)
-        // on macOS deleted file will not be in the ./Trash directory, but it doesn't mean that it is not removed because it is placed in the NSMetadataQueryUpdateRemovedItems array.
-        item.isRemoved = true
-        return item
-      } catch {
-        onError(error)
-        return nil
-      }
-    })
+  class func getTrashContentsFromNotification(_ notification: Notification) throws -> CloudContents {
```

Comment:
Which code/event moves files into the trash (on iOS and Mac)? Or is it only to handle explicit user-initiated deletion of .kml using Files app?

Reply:
1. Regular file deletion deletes files from the `bookmarks` and trashes the cloud file (move from icloud to the icloud/.trash by using filemanager default method):
https://github.com/organicmaps/organicmaps/blob/2f8f9ded88172781060ac3f71f68d3f52bab021a/iphone/Maps/Core/iCloud/SynchronizationFileWriter.swift#L133

2. And explicit deletion from the Files app.

Reply:
What is "Regular file deletion"? Is it when user deletes a bookmarks list in Organic Maps?

It is clear that deletion from Files app puts deleted files into the trash. It is not clear how deleting bookmarks in OM puts them into the trash...